src/forklift/core.py

```diff
@@ -184,9 +185,19 @@ def abstract_type(type):
             if abstract_type(source_fld.type) != abstract_type(destination_fld.type):
                 mismatching_fields.append('{}: source type of {} does not match destination type of {}'
                                           .format(source_fld.name, source_fld.type, destination_fld.type))
-            elif source_fld.type == 'String' and source_fld.length != destination_fld.length:
-                mismatching_fields.append('{}: source length of {} does not match destination length of {}'
-                                          .format(source_fld.name, source_fld.length, destination_fld.length))
+            elif source_fld.type == 'String':
+                def truncate_field_length(field):
+                    if field.length > 4000:
+                        log.warn('%s is longer than 4000 characters. Truncation may occur.', field.name)
+                        return 4000
+                    else:
+                        return field.length
+
+                source_fld.length = truncate_field_length(source_fld)
+                destination_fld.length = truncate_field_length(destination_fld)
```

Comment:
Just for standardization of the comparison in the line below? I don't know, really.